sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -667,3 +671,26 @@ def test_mutual_info_regression():
     gtruth = np.zeros(10)
     gtruth[:2] = 1
     assert_array_equal(support, gtruth)
+
+
+def test_univariate_nan_inf_allowed_in_transform():
+    X, y = make_regression(n_samples=100, n_features=10, n_informative=2,
+                           shuffle=False, random_state=0, noise=10)
+
+    univariate_filter = GenericUnivariateSelect(f_regression,
+                                                mode='percentile')
+    univariate_filter.fit(X, y)
```

Comment:
we could fit with nan / inf too and remove `test_univariate_nan_inf_allowed_in_fit`

Reply:
@NicolasHug the reason they were separate was that fit can only be tested with the dummy scorer because all of the actual scorers don't allow nan/inf. I feel like it's a "stronger" test to test transform with nan/inf with a real scorer in use, rather than just the dummy one. Maybe it doesn't matter though and I could combine them both to just use the dummy_score?